traitsui/tests/ui_editors/test_data_frame_editor.py

```diff
@@ -88,411 +87,409 @@ def sample_text_data():
     return viewer
 
 
-@skip_if_null
-def test_adapter_get_item():
-    viewer = sample_data()
-    adapter = DataFrameAdapter()
+class TestDataFrameEditor(unittest.TestCase):
 
-    item_0_df = adapter.get_item(viewer, "data", 0)
+    @skip_if_null
```

Comment:
Not a request change, just acknowledging an existing issue:
Looks like the GUI event loop is not required for this test, I am wondering if this skip is necessary at all.
That said, this is orthogonal to the purpose of this PR so it would also make sense for this to be changed separately.

Reply:
My suspicion is that because the `DataFrameEditor` is really a TraitsUI `View` wrapped around a `TabularEditor` and inserted into a special editor type, that it requires more TraitsUI machinery than is available in the `Null` editor - it's not a question of event loops, its that the null backend doesn't implement things (not even dummy versions).

Reply:
On master, if I remove the `skip_if_null` for tests that don't call `edit_traits`, I get this:
```
$ python -m nose.core -v traitsui.tests.ui_editors.test_data_frame_editor
/Users/kchoi/.edm/envs/traitsui-test-3.6-null/lib/python3.6/runpy.py:125: RuntimeWarning: 'nose.core' found in sys.modules after import of package 'nose', but prior to execution of 'nose.core'; this may result in unpredictable behaviour
  warn(RuntimeWarning(msg))
traitsui.tests.ui_editors.test_data_frame_editor.test_adapter_get_item ... ok
traitsui.tests.ui_editors.test_data_frame_editor.test_adapter_empty_dataframe ... ok
traitsui.tests.ui_editors.test_data_frame_editor.test_adapter_no_rows ... ok
traitsui.tests.ui_editors.test_data_frame_editor.test_adapter_get_item_numerical ... ok
traitsui.tests.ui_editors.test_data_frame_editor.test_data_frame_editor ... SKIP: Test not working on the 'null' backend
traitsui.tests.ui_editors.test_data_frame_editor.test_data_frame_editor_alternate_adapter ... SKIP: Test not working on the 'null' backend
traitsui.tests.ui_editors.test_data_frame_editor.test_data_frame_editor_columns ... SKIP: Test not working on the 'null' backend
traitsui.tests.ui_editors.test_data_frame_editor.test_data_frame_editor_font_mapping ... SKIP: Test not working on the 'null' backend
traitsui.tests.ui_editors.test_data_frame_editor.test_data_frame_editor_format_mapping ... SKIP: Test not working on the 'null' backend
traitsui.tests.ui_editors.test_data_frame_editor.test_data_frame_editor_multi_select ... SKIP: Test not working on the 'null' backend
traitsui.tests.ui_editors.test_data_frame_editor.test_data_frame_editor_numerical_index ... SKIP: Test not working on the 'null' backend
traitsui.tests.ui_editors.test_data_frame_editor.test_data_frame_editor_text_data ... SKIP: Test not working on the 'null' backend
traitsui.tests.ui_editors.test_data_frame_editor.test_data_frame_editor_with_refresh ... SKIP: Test not working on the 'null' backend
traitsui.tests.ui_editors.test_data_frame_editor.test_data_frame_editor_with_update_refresh ... SKIP: Test not working on the 'null' backend
traitsui.tests.ui_editors.test_data_frame_editor.test_adapter_delete_start ... ok
traitsui.tests.ui_editors.test_data_frame_editor.test_adapter_delete_start_numerical_index ... ok
traitsui.tests.ui_editors.test_data_frame_editor.test_adapter_delete_middle ... ok
traitsui.tests.ui_editors.test_data_frame_editor.test_adapter_delete_middle_numerical_index ... ok
traitsui.tests.ui_editors.test_data_frame_editor.test_adapter_delete_end ... ok
traitsui.tests.ui_editors.test_data_frame_editor.test_adapter_delete_end_numerical_index ... ok
traitsui.tests.ui_editors.test_data_frame_editor.test_adapter_insert_start ... ok
traitsui.tests.ui_editors.test_data_frame_editor.test_adapter_insert_start_numerical_index ... ok
traitsui.tests.ui_editors.test_data_frame_editor.test_adapter_insert_middle ... ok
traitsui.tests.ui_editors.test_data_frame_editor.test_adapter_insert_middle_numerical_index ... ok
traitsui.tests.ui_editors.test_data_frame_editor.test_adapter_insert_end ... ok
traitsui.tests.ui_editors.test_data_frame_editor.test_adapter_insert_end_numerical_index ... ok
traitsui.tests.ui_editors.test_data_frame_editor.test_adapter_set_text ... ok
traitsui.tests.ui_editors.test_data_frame_editor.test_adapter_set_text_invalid ... ok
traitsui.tests.ui_editors.test_data_frame_editor.test_adapter_set_index_text ... ok
traitsui.tests.ui_editors.test_data_frame_editor.test_adapter_set_index_text_numeric ... ok
traitsui.tests.ui_editors.test_data_frame_editor.test_adapter_set_index_text_numeric_invalid ... ok
```

I will open an issue to remove these redundant skips. It is orthogonal to this PR.

Reply:
https://github.com/enthought/traitsui/issues/816